pyparsing/diagram/__init__.py

```diff
@@ -379,6 +401,12 @@ def _to_diagram_element(
         number=lookup.generate_index(),
     )
 
+    if _is_named_terminal(element):
```

Comment:
So here you're bypassing the normal mechanism of "only extract if we've seen it twice". Is this intentional? I forget what we discussed. If so, there's probably an easier way to do this than using the extraction mechanism at all, but I can't think of exactly what it might be.

Reply:
Yeah, I think this is intentional. Even if an object is only seen once, if it is sufficiently complex (and has a name), it should be a separate subdiagram. This makes the diagrams more likely to map to a BNF.